docs/architecture/data-plane-osd-cluster-dynamic-scaling.md

```diff
@@ -6,18 +6,181 @@ This documents explains how Data Plane OSD Cluster Dynamic Scaling functionality
+    the provider's region has not been reached
+ 2. There is no scale up action ongoing. A scale up action is ongoing
+    if there is at least one cluster in the following states: `provisioning`,
+    `provisioned`, `accepted`, `waiting_for_kas_fleetshard_operator`
+ 3. At least one of the two following conditions are true:
+      * No cluster in the provider's region has enough capacity to allocate
+        the biggest instance size of the given instance type
+      * The free capacity (in streaming units) for the given instance type in
+        the provider's region is smaller or equal than the defined slack
+        capacity (also in streaming units) of the given instance type. Free
+        capacity is defined as max(total) capacity - consumed capacity.
+        For the calculation of the max capacity:
+        * Clusters in `deprovisioning` and `cleanup` state are excluded, as
+          clusters into those states don't accept kafka instances anymore.
+        * Clusters that are still not `ready` to accept kafka instance but that
+          should eventually accept them (like accepted state for example)
+          are included
+          
+>NOTE: cluster in `failed` state are not counted in capacity and limit calculations.
+
+#### OSD cluster creation and terraforming
+
+Once the fleet manager has evaluated that there is a need to create a cluster in a given region that supports a given instance type, it will proceed on creating a new cluster that has the following characteristics:
+- `supported_instance_type` will be set to the Streaming unit instance type as evaluated in the scale up detection logic
+- `cloud_provider` and `region` will be set to the cloud provider and region that's needs scaling up
+- `provider_type` is harcoded to `ocm` as it is the only cluster provider that's support auto creation
+- `status` is set to `accepted`
+- `multi_az` will be set to `true` if the supported instance type is `standard`. Otherwise, it is set to `false` for developer supporting clusters
+ 
+During provisioning phase, the default machine pool will be autoscaled and the `max` number of nodes is hardcoded to `18` while the minimum is hardcoded to `6`. The compute machine type is read from the configuration value. 
+
+Once the cluster has been provisioned, dedicated machine pools will be created par supported instance type.
+The created machine pools will be autoscaled: the minimum number of nodes is set to `1` if the cluster is single az. Otherwise it is set to `3`. 
+The maximum number of nodes is the one given by the [dynamic_scaling_config.go](../../internal/kafka/internal/config/dynamic_scaling_config.go)
+Once the machine pools have been created, the `max_nodes` for each supported instance type will be stored in the database.
+
+Once the cluster has been terraformed, Fleetshard sync will send back the capacity information and they'll be stored in the database.
+
+### OSD cluster deletion
+
+#### OSD cluster deletion evaluation
```

Comment:
This might need modification after the implementation is done and refined

Reply:
+1